### ISSUE TITLE
Incorrect "previous iteration of loop" error message

### ISSUE THREAD

**Issue by grantslatton** [rust-lang/rust#69275]
The compiler gives a "value moved in previous iteration of loop" error message where there is no loop present.

Code: [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=b90242be8f2df1b021024dba858d0817):

```rust
fn main() {
    macro_rules! foo {
        ($x:expr) => {
            let _y = $x;
            let _z = $x;
        }
    }
    
    let non_copy_value = vec![0u8];
    foo!(non_copy_value);
}
```

I expected to see an error message that does not talk about loops.

Instead, the compiler gives this error message: 

```
error[E0382]: use of moved value: `non_copy_value`
  --> src/main.rs:11:10
   |
10 |     let non_copy_value = vec![0u8];
   |         -------------- move occurs because `non_copy_value` has type `std::vec::Vec<u8>`, which does not implement the `Copy` trait
11 |     foo!(non_copy_value);
   |          ^^^^^^^^^^^^^^ value moved here, in previous iteration of loop

```

### Meta
Bug exists on stable 1.41.0 and still on nightly 1.43.0

**Comment by grantslatton** [comment]
This appears to be fixed on stable now. Current output is:

```
error[E0382]: use of moved value: `non_copy_value`
  --> src/main.rs:11:10
   |
10 |     let non_copy_value = vec![0u8];
   |         -------------- move occurs because `non_copy_value` has type `Vec<u8>`, which does not implement the `Copy` trait
11 |     foo!(non_copy_value);
   |          ^^^^^^^^^^^^^^
   |          |
   |          value moved here
   |          value used here after move
```

Does this need a test before closing or can we just close it?

**Comment by estebank** [comment]
Current output

```
error[E0382]: use of moved value: `non_copy_value`
  --> src/main.rs:11:10
   |
10 |     let non_copy_value = vec![0u8];
   |         -------------- move occurs because `non_copy_value` has type `Vec<u8>`, which does not implement the `Copy` trait
11 |     foo!(non_copy_value);
   |          ^^^^^^^^^^^^^^ value used here after move
   |
help: consider cloning the value if the performance cost is acceptable
   |
11 |     foo!(non_copy_value.clone());
   |                        ++++++++
```

I'm pretty sure we already have tests for this in the test suite :)
